File: 485-max-consecutive-ones/max-consecutive-ones.ts
function findMaxConsecutiveOnes(nums: number[]): number {
    let maxOnes: number = 0
    let currentOnes: number = 0

    for (let i = 0; i < nums.length; i++) {
         if (nums[i] === 1) {
            currentOnes++
            maxOnes = Math.max(currentOnes, maxOnes)
         } else {
            currentOnes = 0
         }
    }

    return maxOnes
};

File: 485-max-consecutive-ones/max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        max_ones = 0
        count = 0

        for right in range(n):
            if nums[right] == 1:
                count += 1
                max_ones = max(max_ones, count)
            else:
                count = 0
                left = right + 1
                
            right += 1

        return max_ones